Add Gradio Demo for weather agent example

freddyaboulton opened the pull request:
The only chat app UI in the docs involves knowledge of html/ts and sqlite.

I figured there is value in showing pydantic ai developers that they can build customizable chat apps with their agents all in python.

sydney-runkle commented:
Hey @freddyaboulton!

We'd love to include something like this.

Give us a little bit more time to iron out bugs like https://github.com/pydantic/pydantic-ai/issues/363 and https://github.com/pydantic/pydantic-ai/issues/268, which cause this application to fail unexpectedly.

freddyaboulton commented:
Hi @sydney-runkle ! I noticed that the demo I added got out of synch with the latest changes in pydantic-ai. So I have updated it. Here is how it's looking!

![pydantic_ai_agent](https://github.com/user-attachments/assets/944b7ccb-1c5a-46c9-8e72-1b80758df185)

sydney-runkle commented:
Fantastic, will stand this up locally and give it a review soon.

Thanks for your patience over the holidays!

sydney-runkle commented on docs/examples/weather-agent.md:
```diff
@@ -5,6 +5,7 @@ Demonstrates:
 * [tools](../tools.md)
 * [agent dependencies](../dependencies.md)
 * [streaming text responses](../results.md#streaming-text)
+* Building a Gradio UI for the agent
```
Let's add a link to the gradio docs :)

sydney-runkle commented on docs/examples/weather-agent.md:
```diff
@@ -28,3 +29,18 @@ python/uv-run -m pydantic_ai_examples.weather_agent
+
+## Running the UI
+
+You can build multi-turn chat applications for your agent with Gradio
```
Again, gradio link please

sydney-runkle replied:
And maybe give a small explanation of what gradio is

sydney-runkle commented on pyproject.toml:
```diff
@@ -132,6 +132,7 @@ venvPath = ".venv"
 executionEnvironments = [
   { root = "tests", reportUnusedFunction = false },
 ]
+exclude = ["examples/pydantic_ai_examples/weather_agent_gradio.py"]
```
Why is this necessary?

sydney-runkle replied:
I see, gradio is not really pyright friendly :(

sydney-runkle reviewed (commented):
Hi @freddyaboulton,

Thanks for your contribution! Left a few initial comments. Could you please make sure your `uv.lock` is up to date with `main`? Thanks!

Once that's done, I'll do some more testing locally :)

freddyaboulton commented:
Hello @sydney-runkle ! I have brought the lockfile up to date and made the changes you suggested. Should be good for a deeper review now :)

sydney-runkle commented:
Great, thanks! I'm reviewing + testing locally now, then will get this across the line shortly 👍

sydney-runkle reviewed (approved):
Thanks for your contribution! I was able to get this up and running locally quite easily.

freddyaboulton commented:
Thank you @sydney-runkle ! happy to contribute for Agent UIs in the future as well!

sydney-runkle commented:
We're working on converting the gif to a video, then will merge this :).
